core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -66,6 +69,8 @@
     private final List<VirtualHost> virtualHosts;
     @Nullable
     private final Mapping<String, VirtualHost> virtualHostMapping;
+    @Nullable
+    private final Map<Integer, Mapping<String, VirtualHost>> virtualHostAndPortMapping;
```

Comment:
- How about merging these two mappings into one? Could use `-1` as a port number for the vhosts without a port.
- Could use a map in fastutil to avoid boxing?

Reply:
Makes a lot of sense. 👍